components/Swap/StepsComponent.tsx

```diff
@@ -43,20 +43,50 @@ export default function Steps({ steps }) {
                                         </span>
                                     </div>
                                 </>
+                            ) : step?.status === 'failed' ? (
+                                <>
+                                    {stepIdx !== steps.length - 1 ? (
+                                        <div className="absolute top-1/2 left-4 -ml-px mt-0.5 h-[40%] w-0.5 bg-gray-300 opacity-60" aria-hidden="true" />
+                                    ) : null}
+                                    <div className="group relative flex items-start">
+                                        <span className="flex h-9 items-center" aria-hidden="true">
+                                            <XCircle className="h-8 w-8 text-red-600" aria-hidden="true" />
+                                        </span>
+                                        <span className="ml-4 flex min-w-0 flex-col">
+                                        <span className="text-sm font-medium text-gray-300">{step.name}</span>
+                                            <span className="text-sm text-primary-text">{step?.description}</span>
+                                        </span>
+                                    </div>
+                                </>
+                            ) : step?.status === 'user_transfer_delayed' ? (
```

Comment:
Don't have a check on `user_transfer_delayed`. Add a more generic term like `delayed` that can later also be used for other cases + will decouple this component from swap's status.